metadata/src/main/java/com/redhat/lightblue/metadata/FieldConstraint.java

```diff
@@ -31,4 +31,10 @@
      * Determines if the constraint is valid for the given field type
      */
     boolean isValidForFieldType(Type fieldType);
+
+
+    /**
+     * Returns the description of the FieldConstraint (for documentation propose)
+     */
+    String describeConstraint();
```

Comment:
This name was confusing when I first saw it.  Maybe if it were `getDescription()` it would be clearer?